README.md

```diff
@@ -1,118 +1,167 @@
+1. [Create authentication credentials](https://techdocs.akamai.com/developer/docs/set-up-authentication-credentials).
+   
+2. Place your credentials in an EdgeGrid resource file, `.edgerc`, under a heading of `[default]` at your local home directory or the home directory of a web-server user.
+   
+   ```
+   [default]
+    client_secret = C113nt53KR3TN6N90yVuAgICxIRwsObLi0E67/N8eRN=
+    host = akab-h05tnam3wl42son7nktnlnnx-kbob3i3v.luna.akamaiapis.net
+    access_token = akab-acc35t0k3nodujqunph3w7hzp7-gtm6ij
+    client_token = akab-c113ntt0k3n4qtari252bfxxbsl-yvsdj
+   ```
+   
+3. Use `egcurl` to sign your requests along with and `--eg-edgerc` argument to point to the path of your `.edgerc` configuration file and an `--eg-section` argument to specify the credentials' section header.
+   
+   ```shell
+   egcurl --eg-edgerc ~/.edgerc --eg-section default --request GET
+   ```
 
+### `~/.egcurl` configuration
 
-## INSTALLATION
+`egcurl` supports an older non-standard configuration file, specified by `--eg-config`. If you use this argument or if the `~/.egcurl` file exists, this configuration file will be consulted first to retrieve your client configuration. If the file doesn't exist or the section isn't found, `egcurl` will look for the client configuration in your `~/.edgerc` file automatically.
 
-1. Install python 2.7 or newer. If you are running GNU/Linux or Mac OS X, you probably already have it.
-2. Install curl. The script expects to find it in your path.
-3. Install edgegrid-python. This is used to sign requests. (`pip install edgegrid-python`)
-4. Clone this repository and then you can execute egcurl directly from the repository clone.
+The older `~/.egcurl` configuration is **deprecated**. Support for this file and format will be removed on or shortly after 2017-05-01. You can easily convert an older `~/.egcurl` configuration to an `~/.edgerc` with the `convert_egcurl.pl` script. For example:
 
+```
+$ ./convert_egcurl.pl < ~/.egcurl > ~/.edgerc
+$ mv ~/.egcurl ~/.egcurl-backup
+```
 
-## CONFIGURATION
+## Use
 
-The EdgeGrid plugin relies on an `~/.edgerc` credentials file that needs to be created in your home directory and organized by [section] following the format below. Each [section] can contain a different credentials set allowing you to store all of your credentials in a single `~/.edgerc` file.
+To use the library, provide the credential's section header of your `.edgerc` file and the appropriate endpoint information.
 
-```
-    [default]
-    client_secret = xxxx
-    host = xxxx # Note, don't include the https:// here
-    access_token = xxxx
-    client_token = xxxx
-    max-body = xxxx
-
-    [section1]
-    client_secret = xxxx
-    host = xxxx # Note, don't include the https:// here
-    access_token = xxxx
-    client_token = xxxx
-    max-body = xxxx
+```shell
+egcurl --eg-edgerc ~/.edgerc --eg-section default --request GET \
+     --url /identity-management/v3/user-profile \
```

Comment:
Wouldn't it be better to add "https://$hostname"? This is what we have on Akamai documentation in Recipes, for example here https://techdocs.akamai.com/property-mgr/recipes/list-cp-codes

Reply:
I think the downside of starting with "https://$host/" is that it will make the reader ask where `$host` is coming from. I worry that will be confusing. I'm hoping to update egcurl soon so that the scheme and host are totally optional, then we can update the examples to exclude them too.